themes/command_duration.theme.bash

```diff
@@ -1,40 +1,33 @@
 # shellcheck shell=bash
 
-if [ -z "$BASH_IT_COMMAND_DURATION" ] || [ "$BASH_IT_COMMAND_DURATION" != true ]; then
+if [[ "${BASH_IT_COMMAND_DURATION:-false}" != true ]]; then
 	_command_duration() {
 		echo -n
 	}
 	return
 fi
 
-# Define tmp dir and file
-COMMAND_DURATION_TMPDIR="${TMPDIR:-/tmp}"
-COMMAND_DURATION_FILE="${COMMAND_DURATION_FILE:-$COMMAND_DURATION_TMPDIR/bashit_theme_execution_$BASHPID}"
+COMMAND_DURATION_START_TIME=
 
 COMMAND_DURATION_ICON=${COMMAND_DURATION_ICON:-'  '}
 COMMAND_DURATION_MIN_SECONDS=${COMMAND_DURATION_MIN_SECONDS:-'1'}
 
-trap _command_duration_delete_temp_file EXIT HUP INT TERM
-
-_command_duration_delete_temp_file() {
-	if [[ -f "$COMMAND_DURATION_FILE" ]]; then
-		rm -f "$COMMAND_DURATION_FILE"
-	fi
-}
-
 _command_duration_pre_exec() {
-	date +%s.%1N > "$COMMAND_DURATION_FILE"
+	local command_nano_now="$(date +%1N)"
+	[[ "$command_nano_now" == "1N" ]] && command_nano_now=1
+	COMMAND_DURATION_START_TIME="$(date "+%s").${command_nano_now}"
 }
 
 _command_duration() {
 	local command_duration command_start current_time
 	local minutes seconds deciseconds
 	local command_start_sseconds current_time_seconds command_start_deciseconds current_time_deciseconds
-	current_time=$(date +%s.%1N)
+	local command_nano_now="$(date +%1N)"
```

Comment:
you duplicate this code twice- you can move that to an external function

Reply:
I'd much rather remove the calls to date entirely, but that would remove the nano-second precision on Linux. Does anyone need nano-second precision? This whole calculation would be much simpler to just use $SECONDS.